### PR TITLE
 Merge m3msg into monorepo

### PR DESCRIPTION
**Pull request by cw9**

### CODE REVIEW

File: .buildkite/pipeline.yml [hunk 1]
```diff
@@ -54,7 +54,7 @@ steps:
       docker-compose#v2.5.1:
         run: app
     <<: *common
-  - label: "Integration (collector, m3em, agg)"
+  - label: "Integration (collector, m3em, agg, msg)"
```
Comment (prateek) [inline .buildkite/pipeline.yml]: do you want to add `test-ci-integration-msg` too

**Comment by prateek** [comment]
still have a few failures in the code, e.g. 
```
github.com/m3db/m3/src/collector/reporter/m3aggregator]: exit status 1
2018-10-12T18:01:03.060600372Z src/collector/reporter/m3aggregator/reporter.go:31:2: cannot find package "github.com/m3db/m3aggregator/client" in any of:
2018-10-12T18:01:03.060604762Z 	/go/src/github.com/m3db/m3/vendor/github.com/m3db/m3aggregator/client (vendor tree)
2018-10-12T18:01:03.060608361Z 	/usr/local/go/src/github.com/m3db/m3aggregator/client (from $GOROOT)
```

also need to rewire the msg codegen targets to use the monorepo tooling

File: .buildkite/pipeline.yml (hunk 1 above)
Comment (cw9) [reply .buildkite/pipeline.yml]: sure

File: glide.lock [hunk 2]
```diff
@@ -292,16 +287,6 @@ imports:
   - x/bytes
 - name: github.com/m3db/m3msg
   version: bb2c675f58f01d487fc0a64aa6c864bc1cc541c9
-  subpackages:
```
[the remark this answers is not in the document]
Comment (cw9) [reply glide.lock]: sure

**Comment by cw9** [comment]
> still have a few failures in the code, e.g.
> 
> ```
> github.com/m3db/m3/src/collector/reporter/m3aggregator]: exit status 1
> 2018-10-12T18:01:03.060600372Z src/collector/reporter/m3aggregator/reporter.go:31:2: cannot find package "github.com/m3db/m3aggregator/client" in any of:
> 2018-10-12T18:01:03.060604762Z 	/go/src/github.com/m3db/m3/vendor/github.com/m3db/m3aggregator/client (vendor tree)
> 2018-10-12T18:01:03.060608361Z 	/usr/local/go/src/github.com/m3db/m3aggregator/client (from $GOROOT)
> ```
> also need to rewire the msg codegen targets to use the monorepo tooling

I think master is failing, trying to fix it now

**Pull request merged**